qemu/tests/blockdev_snapshot_install.py

```diff
@@ -0,0 +1,60 @@
+import time
+import logging
+import random
+
+from virttest import utils_test
+from virttest import utils_misc
+from virttest.tests import unattended_install
+
+from provider.blockdev_snapshot_base import BlockDevSnapshotTest
+
+
+def run(test, params, env):
+    """
+    Backup VM disk test when VM reboot
+
+    1) start VM with system disk
+    2) create target disk with qmp command
+    3) load stress in guest
+    4) do snapshot to target disk
+    5) shutdown VM
+    6) boot VM with target disk
+    :param test: test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    def tag_for_install(vm, tag):
+        if vm.serial_console:
+            serial_output = vm.serial_console.get_output()
+            if tag in serial_output:
+                return True
+        logging.info("Failed to start installation process")
+        return False
+
+    base_image = params.get("images", "image1").split()[0]
+    params.update(
+        {"image_name_%s" % base_image: params["image_name"],
+         "image_format_%s" % base_image: params["image_format"]})
+    snapshot_test = BlockDevSnapshotTest(test, params, env)
+    args = (test, params, env)
+    bg = utils_test.BackgroundTest(unattended_install.run, args)
+    bg.start()
+    if bg.is_alive():
+        tag = params["tag_for_install_start"]
+        if utils_misc.wait_for(lambda: tag_for_install(snapshot_test.main_vm, tag), 120, 10, 5):
+            logging.info("sleep random time before do snapshots")
+            time.sleep(random.randint(120, 600))
+            snapshot_test.pre_test()
+            try:
+                snapshot_test.create_snapshot()
+                try:
+                    bg.join(timeout=1200)
+                except Exception:
+                    raise
+                snapshot_test.verify_snapshot()
+            finally:
+                snapshot_test.post_test()
```

Comment:
logs after rebase:

    -blockdev node-name=file_sn1,driver=file,aio=threads,filename=/home/kvm_autotest_root/images/sn1.qcow2,cache.direct=on,cache.no-flush=off \
    -blockdev node-name=drive_sn1,driver=qcow2,cache.direct=on,cache.no-flush=off,file=file_sn1 \
    -device scsi-hd,id=sn1,drive=drive_sn1,write-cache=on \
    -device virtio-net-pci,mac=9a:96:68:58:09:d9,id=idPGz7Wt,netdev=id7SjRcL,bus=pci.0,addr=0x5  \
    -netdev tap,id=id7SjRcL,vhost=on,vhostfd=24,fd=17 \
    -blockdev node-name=file_cd1,driver=file,read-only=on,aio=threads,filename=/home/kvm_autotest_root/iso/linux/RHEL-8.2.0-20200404.0-x86_64-dvd1.iso,cache.direct=on,cache.no-flush=off \
    -blockdev node-name=drive_cd1,driver=raw,read-only=on,cache.direct=on,cache.no-flush=off,file=file_cd1 \
    -device scsi-cd,id=cd1,drive=drive_cd1,write-cache=on \
    -blockdev node-name=file_unattended,driver=file,read-only=on,aio=threads,filename=/home/kvm_autotest_root/images/rhel820-64/ks.iso,cache.direct=on,cache.no-flush=off \
    -blockdev node-name=drive_unattended,driver=raw,read-only=on,cache.direct=on,cache.no-flush=off,file=file_unattended \
    -device scsi-cd,id=unattended,drive=drive_unattended,write-cache=on  \
    -kernel '/home/kvm_autotest_root/images/rhel820-64/vmlinuz'  \
    -append 'inst.sshd ksdevice=link inst.repo=cdrom:/dev/sr0 inst.ks=cdrom:/dev/sr1:/ks.cfg nicdelay=60 biosdevname=0 net.ifnames=0 console=ttyS0,115200 console=tty0'  \
    -initrd '/home/kvm_autotest_root/images/rhel820-64/initrd.img'  \

As you can see that, another installation began? what we want is booting VM from sn1, but here we cannot make sure VM boots successfully(booting from cd? ).

Reply:
It's not another installation, it's a normal boot process, see cmdline "-boot menu=off,order=cdn,once=c,strict=off", it means boot from hard disk first, if boot failed, then will boot from cdrom.

Reply:
The same question, how to make sure that vm boots from sn1, given the situation that booting from sn1 failed, then vm tried to boot from cd, no error occurred

Reply:
For "-boot menu=off,order=cdn,once=c,strict=off", it means:
If sn1 is empty:
    It will first try to boot from sn1, if failed, then boot from cdrom, then start guest installation process.
else:
    It will boot from sn1 even though data on sn1 is not integral.

Here, we do snapshot during guest installation, so there must be data on sn1.
So even if we start guest with sn1 and installation options set, it will just only  boot from disk.

Will update the code to make verify of sn1's boot more effective.

Reply:
IMO, even if the image is full of data, vm can still fail to boot from the image if  it is not a bootable device, and in this case, it's an assertion to be verified if 'sn1(+backing)' is bootable or not.